src/main/java/com/app/eece4792mealmaster/services/StockService.java

```
@@ -107,6 +107,12 @@ public boolean deleteStockItem(Long stockItemId) {
     return toDelete.isPresent();
   }
 
+  public boolean deleteFoodStock(Long foodStockId) {
+    Optional<FoodStock> toDelete = foodStockRepository.findById(foodStockId);
+    toDelete.ifPresent(stockItem -> foodStockRepository.delete(stockItem));
+    return toDelete.isPresent();
```

return !toDelete.isPresent(). It failed if it's still there

---

actually will toDelete show the correct presence if you instantiate it before deleting? or do you have to findById again?

---

I thought this returns true if it was found

---

if it doesn't successfully delete, that is like a database exception, which is a huge deal. I think it should throw an exception for that instead of a boolean, which I think it does if the repository doesn't succeed, right?

---

Not sure. You're probably right though